### ISSUE TITLE
Wording question: What do you call endless loops

### ISSUE THREAD

**Issue by codepunkt** [redux-saga/redux-saga#861]
I've seen sagas that run in a loop (e.g. `while (true)` or `while (yield take(x))` or `takeEvery(x)`) being described/named a lot of things, thus creating ambiguity in conversations.

- loop
- daemon
- flow
- watcher

what is your stance on the matter? is there a preferred term for these?

**Comment by Andarist** [comment]
I think the most popular term is a watcher, although aint sure if its
appropriate for endless loops which dont include take

**Comment by granmoe** [comment]
I think this is a great question. Some would call it bikeshedding, but we really struggled with this on a recent project. We have over 60 watchers, and when we started using saga, we didn't know what to call them. We decided (or rather, I did) to call the watchers "sagas," but call sagas that are completely contained in their own function (no separate watcher and worker) "flows." I figured that the things that you pass into the middleware are strictly speaking "sagas," and so I thought this would make sense. However, it quickly became super confusing and not everyone could tell from the naming conventions what kind of function each thing was. So we ditched that pattern because it's simply way too vague and confusing in a large project. We've found that the watcher/worker convention is the most clear, even given the caveat that it might not fit 100% of all possible scenarios. We still call sagas that have their watcher and worker all in one function "sagas" (like "submitPaymentSaga" for example). 

So watcher/worker is most popular in the community, but that's not to say someone couldn't come up with a more clear naming convention.
